Reely/Controller/FilmViewController.swift

```diff
@@ -0,0 +1,248 @@
+    var suggestionsController : FilmSearchController!
+    var resultsSearchController: UISearchController!
+    @IBOutlet weak var filmTableView: UITableView!
+    
+    /* custom font  */
+    let GillSans = UIFont(name: "Gill Sans", size: UIFont.labelFontSize)
+    
+    /* custom tint color  */
+    let plum = hexStringToUIColor(hex: "#554660")
+    
+    
+    // MARK: - The Child View Controller
+    private lazy var filmSearchController: FilmSearchController = {
+        // Load Storyboard
+        let storyboard = UIStoryboard(name: "Main", bundle: Bundle.main)
+        
+        // Instantiate View Controller
+        var viewController = storyboard.instantiateViewController(withIdentifier: "filmSearchController") as! FilmSearchController
+        
+        // Add View Controller as Child View Controller
+        self.addChild(viewController)
+        viewController.filmViewController = self
+        
+        return viewController
+    }()
+    
+    
+    // MARK: - viewDidLoad()
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        
+       /* remove history for testing
+         previously the app would crash if there was no history*/
+//        UserDefaults.standard.removeObject(forKey: "searches")
+        
+        
+        
+        /* setup the view's controller hierarchy  */ 
+        setupView()
+        
+        
+        /* make sure that the searchArray has 10 slots only  */
+        searchArray.reserveCapacity(10)
+        for _ in 0..<10 {
+            searchArray.append(String())
+        }
+        
+        /* set the tableview's delegate and datasource to this class*/
+        filmTableView.estimatedRowHeight = 250
+        filmTableView.rowHeight = UITableView.automaticDimension
+        
+        /* suggestionsController controls the behavior of the search suggestion tableView  */
+        suggestionsController = FilmSearchController()
+        resultsSearchController = UISearchController(searchResultsController: suggestionsController)
+        
+        loadSearchArray()
+        
+        /* set up the searchController   */
+        self.resultsSearchController = ({
+            let controller = UISearchController(searchResultsController: filmSearchController)
+            
+            /* change the icon of the magnifying glass to white  */
+            let glassIconView = controller.searchBar.searchTextField.leftView as! UIImageView
+            glassIconView.image = glassIconView.image?.withRenderingMode(UIImage.RenderingMode.alwaysTemplate)
+            glassIconView.tintColor = UIColor.white
+            
+            controller.searchResultsUpdater = self
+            controller.searchBar.sizeToFit()
+            controller.view.tintColor = UIColor.white
+            controller.searchBar.searchTextField.placeholder = "Film title here"
+            controller.searchBar.barStyle = UIBarStyle.black
+            controller.searchBar.barTintColor = plum
```

Comment:
this is more of a style guide critique but you can write as `.white` instead of `UIColor.white`

```
glassIconView.image = glassIconView.image?.withRenderingMode(.alwaysTemplate)
glassIconView.tintColor = .white
controller.searchBar.barStyle = .black
```